// @vonage/vvd-foundation/constants.js
export var Connotation;
(function (Connotation) {
    Connotation["Primary"] = "primary";
    Connotation["CTA"] = "cta";
    Connotation["Success"] = "success";
    Connotation["Alert"] = "alert";
    Connotation["Warning"] = "warning";
    Connotation["Info"] = "info";
    Connotation["Announcement"] = "announcement";
})(Connotation || (Connotation = {}));
export var ConnotationDecorative;
(function (ConnotationDecorative) {
    ConnotationDecorative["Pacific"] = "pacific";
})(ConnotationDecorative || (ConnotationDecorative = {}));
export var Shape;
(function (Shape) {
    Shape["Rounded"] = "rounded";
    Shape["Pill"] = "pill";
    Shape["Circled"] = "circled";
})(Shape || (Shape = {}));
export var Size;
(function (Size) {
    Size["xxx_Small"] = "xxxs";
    Size["xx_Small"] = "xxs";
    Size["x_Small"] = "xs";
    Size["Small"] = "sm";
    Size["Medium"] = "md";
    Size["Large"] = "lg";
    Size["x_Large"] = "xl";
    Size["xx_Large"] = "xxl";
    Size["xxx_Large"] = "xxxl";
})(Size || (Size = {}));
export var Layout;
(function (Layout) {
    Layout["Text"] = "text";
    Layout["Filled"] = "filled";
    Layout["Outlined"] = "outlined";
    Layout["Soft"] = "soft";
    Layout["Ghost"] = "ghost";
})(Layout || (Layout = {}));
export var IconSize;
(function (IconSize) {
    IconSize["Small"] = "small";
    IconSize["Medium"] = "medium";
    IconSize["Large"] = "large";
})(IconSize || (IconSize = {}));
export var Position;
(function (Position) {
    Position["Top"] = "TOP";
    Position["Bottom"] = "BOTTOM";
    Position["Start"] = "START";
    Position["Center"] = "CENTER";
    Position["End"] = "END";
})(Position || (Position = {}));
export var Role;
(function (Role) {
    Role["Status"] = "status";
    Role["Alert"] = "alert";
})(Role || (Role = {}));
export var AriaLive;
(function (AriaLive) {
    AriaLive["Polite"] = "polite";
    AriaLive["Assertive"] = "assertive";
})(AriaLive || (AriaLive = {}));
