// @vonage/vivid/shared/enums.js
var Connotation;
(function (Connotation) {
  Connotation["Accent"] = "accent";
  Connotation["CTA"] = "cta";
  Connotation["Success"] = "success";
  Connotation["Alert"] = "alert";
  Connotation["Warning"] = "warning";
  Connotation["Information"] = "information";
  Connotation["Announcement"] = "announcement";
})(Connotation || (Connotation = {}));
var ConnotationDecorative;
(function (ConnotationDecorative) {
  ConnotationDecorative["Pacific"] = "pacific";
})(ConnotationDecorative || (ConnotationDecorative = {}));
var Shape;
(function (Shape) {
  Shape["Rounded"] = "rounded";
  Shape["Pill"] = "pill";
  Shape["Circled"] = "circled";
  Shape["Sharp"] = "sharp";
})(Shape || (Shape = {}));
var Appearance;
(function (Appearance) {
  Appearance["Filled"] = "filled";
  Appearance["Outlined"] = "outlined";
  Appearance["Duotone"] = "duotone";
  Appearance["Fieldset"] = "fieldset";
  Appearance["Subtle"] = "subtle";
  Appearance["Ghost"] = "ghost";
})(Appearance || (Appearance = {}));
var Size;
(function (Size) {
  Size["UltraCondensed"] = "ultraCondensed";
  Size["SuperCondensed"] = "superCondensed";
  Size["ExtraCondensed"] = "extraCondensed";
  Size["SemiCondensed"] = "semiCondensed";
  Size["Condensed"] = "condensed";
  Size["Normal"] = "normal";
  Size["SemiExpanded"] = "semiExpanded";
  Size["Expanded"] = "expanded";
  Size["ExtraExpanded"] = "extraExpanded";
  Size["SuperExpanded"] = "superExpanded";
  Size["UltraExpanded"] = "ultraExpanded";
})(Size || (Size = {}));
var LayoutSize;
(function (LayoutSize) {
  LayoutSize["Small"] = "small";
  LayoutSize["Medium"] = "medium";
  LayoutSize["Large"] = "large";
})(LayoutSize || (LayoutSize = {}));
var Position;
(function (Position) {
  Position["Top"] = "TOP";
  Position["Bottom"] = "BOTTOM";
  Position["Start"] = "START";
  Position["Center"] = "CENTER";
  Position["End"] = "END";
})(Position || (Position = {}));
var Role;
(function (Role) {
  Role["Status"] = "status";
  Role["Alert"] = "alert";
})(Role || (Role = {}));
var AriaLive;
(function (AriaLive) {
  AriaLive["Polite"] = "polite";
  AriaLive["Assertive"] = "assertive";
})(AriaLive || (AriaLive = {}));

export { Connotation as C, Shape as S };
